airflow/contrib/hooks/spark_submit_hook.py

```diff
@@ -492,8 +496,8 @@ def _start_driver_status_tracking(self):
         while self._driver_status not in ["FINISHED", "UNKNOWN",
                                           "KILLED", "FAILED", "ERROR"]:
 
-            # Sleep for 1 second as we do not want to spam the cluster
-            time.sleep(1)
+            # Sleep for n seconds as we do not want to spam the cluster
+            time.sleep(self._status_poll_interval)
```

Comment:
This actually means occupy worker and do nothing for n seconds is it not?
It was OK when it was 1 second but users may set it to even 5 min without realising that it occupys the worker.

My comment here is more of a concern rather than an action to do.
Should this work by occupying the worker "indefinitely" or can it be something like the sensors with (poke/reschedule)?

Reply:
I know what you mean but this is better than existing hardcoding, even changing to sleep 10 will reduce load on spark cluster